07_IPC/03_Message_Passing.md

```diff
@@ -33,16 +33,27 @@ As mentioned, there's some initial setup that goes into message passing: we need
 Since the struct representing the endpoint is going to be accessed by multiple processes, we'll want a lock to protect the data from race conditions. We'll use the following struct to represent our endpoint:
 
 ```c
+
+typedef struct ipc_message{
+    void* message_buffer;
+    size_t message_size;
+    uintptr_t next_message;
+} ipc_message_t;
+
+ typedef struct ipc_message_queue{
```

Comment:
Is this type needed? since an endpoints and queues have a one-to-one relationship I dont think this is necessary. Instead embedding the list head in `ipc_endpoint` should be fine.